### PR TITLE
`BlockId` removal: refactor of runtime API

### PR DESCRIPTION
**Pull request by michalkucharczyk**
It changes the first argument of all generated runtime API calls from: `BlockId<Block>` to: `Block::Hash`

This PR is part of BlockId::Number refactoring analysis (paritytech/substrate#11292)

Companion for: paritytech/substrate#13255
cumulus companion: paritytech/cumulus#2190

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2396985

**Review by bkchr — APPROVED** [review]

File: node/test/client/src/block_builder.rs [hunk 1]
```diff
@@ -88,7 +87,7 @@ impl InitPolkadotBlockBuilder for Client {
 		};
 
 		let mut block_builder = self
-			.new_block_at(&at, digest, false)
+			.new_block_at(&BlockId::Hash(hash), digest, false)
```
Comment (melekes) [inline node/test/client/src/block_builder.rs:90]: I thought you're supposed to pass hash by value, no?

**Review by melekes — APPROVED** [review]
👍

File: node/test/client/src/block_builder.rs (hunk 1 above)
Comment (michalkucharczyk) [reply node/test/client/src/block_builder.rs:90]: Just following the function definition [here](https://github.com/paritytech/substrate/blob/master/client/block-builder/src/lib.rs#L121).

Yes `Hash` and also `BlockId` shall be passed by value, but obviously it is not always the case across substrate modules. Whenever changing the `BlockId` to `Hash` I was introducing by-value API.

**Comment by michalkucharczyk** [comment]
bor rebase

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2415098

**Review by acatangiu — APPROVED** [review]

**Comment by michalkucharczyk** [comment]
bot rebase

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2418913

**Pull request merged**